classes/class-p4-search.php

```diff
@@ -601,14 +630,16 @@ protected function set_results_context( &$context ) {
 						$context['content_types']['3']['results']++;
 				}
 
-				// Page Type <-> Category.
-				$page_types = get_the_terms( $post->ID, 'p4-page-type' );
-				if ( $page_types ) {
-					foreach ( (array) $page_types as $page_type ) {
-						// p4-page-type filters.
-						$context['page_types'][ $page_type->term_id ]['term_id'] = $page_type->term_id;
-						$context['page_types'][ $page_type->term_id ]['name']    = $page_type->name;
-						$context['page_types'][ $page_type->term_id ]['results']++;
+				// Page Type <-> Category. This taxonomy is used only for Posts.
+				if ( 'post' === $post->post_type ) {
+					$page_types = get_the_terms( $post->ID, 'p4-page-type' );
+					if ( $page_types ) {
```

Comment:
@angtheod would it be better if we check that the variable is an array here? Incase get_the_terms returns a WP_Error object.